### PR TITLE
nixos: tailscale: option to disable path warning

### LINKED ISSUE #194623: tailscale module: warning about strict reverse path filtering should be disable-able
### Describe the bug

I really appreciate this message, but also would like to be able to disable it. Most of my Tailscale machines are not exit nodes, nor do they act as subnet routers, hence I would prefer to leave strict filtering enabled and also not have these warnings.

It seems like a small thing, but these warnings tend to make me blind to other new, more relevant warnings.

```
trace: warning: Strict reverse path filtering breaks Tailscale exit node use and some subnet routing setups. Consider setting `networking.firewall.checkReversePath` = 'loose'
```

I'm hacking on something here: https://github.com/colemickens/nixpkgs/tree/tailscale-path-warning

### PR DESCRIPTION
**Pull request by colemickens**
###### Description of changes

fixed #194623 by adding an option `services.tailscale.ignoreReversePathFilter` to quiet a warning that is relevant to only some tailscale clients/users.

cc: @LunNova @danderson 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by LunNova — CHANGES_REQUESTED** [review]

**Comment by colemickens** [comment]
Crap, I fixed this on the wrong branch. Thanks @LunNova

**Comment by colemickens** [comment]
I guess I was fixing it in the right place, cherry-picking it to my usual nixpkgs and just forgot to push. Should be good now.

**Review by LunNova — APPROVED** [review]

**Comment by LunNova** [comment]
@ofborg eval

**Review by danderson — APPROVED** [review]

**Review by LunNova — APPROVED** [review]

**Review by Misterio77 — APPROVED** [review]

File: nixos/modules/services/networking/tailscale.nix [hunk 1]
```diff
@@ -41,15 +48,15 @@ in {
   };
 
   config = mkIf cfg.enable {
-    warnings = optional (firewallOn && rpfIsStrict) "Strict reverse path filtering breaks Tailscale exit node use and some subnet routing setups. Consider setting `networking.firewall.checkReversePath` = 'loose'";
+    warnings = optional (firewallOn && rpfIsStrict && !cfg.ignoreReversePathFilter) "Strict reverse path filtering breaks Tailscale exit node use and some subnet routing setups. Consider setting `networking.firewall.checkReversePath` = 'loose'";
```
Comment (LunNova) [inline nixos/modules/services/networking/tailscale.nix:51]: Might be worth mentioning the option to silence it here?
```suggestion
    warnings = optional (firewallOn && rpfIsStrict && !cfg.ignoreReversePathFilter) "Strict reverse path filtering breaks Tailscale exit node use and some subnet routing setups. Consider setting `networking.firewall.checkReversePath` = 'loose', or silence this warning with `services.tailscale.ignoreReversePathFilter = true;`";
```

**Review by danderson — APPROVED** [review]

File: nixos/modules/services/networking/tailscale.nix (hunk 1 above)
Comment (colemickens) [reply nixos/modules/services/networking/tailscale.nix]: great suggestion! thanks and done

**Comment by SuperSandro2000** [comment]
Do we have any other options to silence warnings?

**Comment by SuperSandro2000** [comment]
I was thinking the same thats why I am not that keen about adding this.

**Comment by ncfavier** [comment]
Let me just mention that there are [discussions upstream](https://github.com/tailscale/tailscale/issues/3310) about fixing the underlying problem so that we eventually wouldn't need a warning.

**Comment by LunNova** [comment]
Fine with this not getting in if it's too hacky. Would a more thought out system for silencing warnings need an RFC?

To solve the warning being annoying locally I could just patch it out anyway.

**Comment by LunNova** [comment]
Existing RFC which discusses adding a way to disable warnings, RFC 0127:

https://github.com/NixOS/rfcs/pull/127/files#diff-e089f932103f6decfd159bd39420f43fd834106b788fa06956e771ffd0cda991R63

**Comment by LunNova** [comment]
Fixed by #201119

**Pull request closed without merging**